File: cases/taint/scala/local-chain-negative/LocalChainNegative.scala
package dataflowbench

object LocalChainNegative {
  def dfb_source(): Int = { // DFB-SOURCE: local-chain-negative-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: local-chain-negative-sink

  def run(): Unit = {
    val first = dfb_source()
    val second = first // DFB-WITNESS: local-chain-negative-second
    val third = second // DFB-WITNESS: local-chain-negative-third
    dfb_sink(0)
  }
}

File: cases/taint/rust/local-chain-negative/local_chain_negative.rs
fn dfb_source() -> i32 { // DFB-SOURCE: local-chain-negative-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: local-chain-negative-sink

fn run() {
    let first = dfb_source();
    let second = first; // DFB-WITNESS: local-chain-negative-second
    let third = second; // DFB-WITNESS: local-chain-negative-third
    dfb_sink(0);
}
